frontend/README.md

```diff
@@ -56,5 +56,29 @@ yarn format
 Build the release bundle:
 
 ```shell
-yarn build 
+yarn build
+```
+
+## Contributing
+
+### Run before submitting a PR
+
+Before submitting a Pull Request, if you have only changed the frontend, you can run the frontend checks alone:
+
+Run the linter:
+
+```shell
+yarn lint
+```
+
+Run the code formatter:
+
+```shell
+yarn format
+```
+
+Run the typescript compiler checks:
+
+```shell
+tsc --noEmit
```

Comment:
Should there be a note added as to where this comes from? Running this on my mac I get:
`zsh: command not found: tsc`

Reply:
Yeah I agree, I'm not sure how to address it. It comes with installing Typescript. Which isn't actually a dependency of the UI project.
So it's only necessary to run tsc...

Reply:
> It comes with installing Typescript. Which isn't actually a dependency of the UI project.

[It is a `devDependency`](https://github.com/stashapp/stash-box/blob/bae0400549c51ec055920ea326ed736ba8f12987/frontend/package.json#L53), and the solution should be `yarn tsc --noEmit` or creating a separate entry in `scripts` for it.